### PR TITLE
docs(eslint): update documentation for eslint flat config

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

Update documentation for eslint flat config

- [x] Review the ESLint configuration with the team

### CODE REVIEW

File: docs/linter/eslint-plugin/rules/yarnrc-package-extensions-harmonize.md [hunk 1]
```diff
@@ -6,7 +6,29 @@ This rule checks the `.yarnrc.yml` file to ensure that each version defined in t
...
-Example of a ESLint configuration:
+Example of a ESLint Flat configuration:
+
+```javascript
+import yamlParser from 'yaml-eslint-parser';
+import o3rPlugin from '@o3r/eslint-plugin';
+
+export default [
+  {
+    files: ['**/package.json'],
```
Comment (kpanot) [inline docs/linter/eslint-plugin/rules/yarnrc-package-extensions-harmonize.md:17]:
```suggestion
    files: ['yarnrc.y{a,}ml'],
```

File: packages/@o3r/eslint-config/src/rules/typescript/eslint-typescript.cjs [hunk 2]
```diff
@@ -88,7 +88,11 @@ const config = [
       '@typescript-eslint/no-non-null-assertion': 'off',
       '@typescript-eslint/no-parameter-properties': 'off',
       '@typescript-eslint/no-redeclare': 'error',
+      // Why do I turn it to off?
+      // If we keep it false, it should be added to the doc.
       '@typescript-eslint/no-redundant-type-constituents': 'off',
+      // Kilian wanted to set it to error.
```
Comment (kpanot) [inline packages/@o3r/eslint-config/src/rules/typescript/eslint-typescript.cjs:94]: (the purpose of enforcing it to error is to prepare `mjs` migration)

Comment (matthieu-crouzet) [reply packages/@o3r/eslint-config/src/rules/typescript/eslint-typescript.cjs:94]: The question is more is it something we recommend to everyone or only to us ?

Comment (kpanot) [reply packages/@o3r/eslint-config/src/rules/typescript/eslint-typescript.cjs:94]: To everyone, we should all be prepared to `mjs` :)

File: docs/linter/eslint-config.md [hunk 3]
```diff
@@ -1,12 +1,40 @@
 # Otter ESLint Configuration
 
-This package provides three different ESLint configuration:
+## Otter ESLint Flat Configuration
+
+`@o3r/eslint-config` provides two differents ESLint configuration:
+
+- **default** (imported via *@o3r/eslint-config*) : Default Otter ESLint configurations used for any JS and TS files.
```
Comment (cpaulve-1A) [inline docs/linter/eslint-config.md:7]:
```suggestion
- **default** (imported via *@o3r/eslint-config*) : Default Otter ESLint configuration used for any JS and TS files.
```

File: docs/linter/eslint-config.md [hunk 4]
```diff
@@ -1,12 +1,40 @@
...
+- **template** (imported via *@o3r/eslint-config/template*) : Otter ESLint configuration to be applied to Otter based Angular templates.
```
Comment (cpaulve-1A) [inline docs/linter/eslint-config.md:8]:
```suggestion
- **template** (imported via *@o3r/eslint-config/template*) : Otter ESLint configuration to be applied to Otter-based Angular templates.
```

File: docs/linter/eslint-config.md [hunk 5]
```diff
@@ -1,12 +1,40 @@
...
+];
+```
+
+
+## Otter ESLint Legacy Configuration
+
+> [!WARNING]
+> These configurations are deprecated. Please use [FlatConfiguration](#otter-eslint-flat-configuration) instead.
+
+`@o3r/eslint-config-otter` provides three different ESLint configuration:
```
Comment (cpaulve-1A) [inline docs/linter/eslint-config.md:31]:
```suggestion
`@o3r/eslint-config-otter` provides three different ESLint configurations:
```

File: docs/linter/eslint-plugin/rules/json-dependency-versions-harmonize.md [hunk 6]
```diff
@@ -6,7 +6,29 @@ This rule checks each `package.json` to ensure that its dependencies are consist
 
 The linter is dedicated to JSON files and requires the setup of the [jsonc-eslint-parser](https://www.npmjs.com/package/jsonc-eslint-parser).
 
-Example of a ESLint configuration:
+Example of a ESLint Flat configuration:
```
Comment (cpaulve-1A) [inline docs/linter/eslint-plugin/rules/json-dependency-versions-harmonize.md:9]:
```suggestion
Example of an ESLint Flat configuration:
```

File: docs/linter/eslint-plugin/rules/no-inner-html.md [hunk 7]
```diff
@@ -7,7 +7,29 @@ For performance reason, it makes sense for certain application to encourage to u
...
+      '@o3r': o3rPlugin
+    }
+    rules: {
+      '@o3r/no-inner-html': 'warn'
+    }
+  }
+]
+```
+
+Example of a ESLint Legacy configuration:
```
Comment (cpaulve-1A) [inline docs/linter/eslint-plugin/rules/no-inner-html.md:32]:
```suggestion
Example of an ESLint Legacy configuration:
```

File: docs/linter/eslint-plugin/rules/yarnrc-package-extensions-harmonize.md [hunk 8]
```diff
@@ -6,7 +6,29 @@ This rule checks the `.yarnrc.yml` file to ensure that each version defined in t
 
 The linter is dedicated to YAML files and requires the setup of the [yaml-eslint-parser](https://www.npmjs.com/package/yaml-eslint-parser).
 
-Example of a ESLint configuration:
+Example of a ESLint Flat configuration:
```
Comment (cpaulve-1A) [inline docs/linter/eslint-plugin/rules/yarnrc-package-extensions-harmonize.md:9]:
```suggestion
Example of an ESLint Flat configuration:
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
+# Migration to ESLint Flat Configuration
+
+[ESLint RC format](https://eslint.org/docs/latest/use/configure/configuration-files-deprecated) is deprecated in favor of [ESLint FlatConfig](https://eslint.org/blog/2022/08/new-config-system-part-2/).
+To migrate, you can follow their [migration guide](https://eslint.org/docs/latest/use/configure/migration-guide).
+
+On Otter side, an ESLint Flat Configuration is exposed by the [`@o3r/eslint-config`](https://www.npmjs.com/package/@o3r/eslint-config) package.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:6]:
```suggestion
On Otter side, an ESLint flat configuration is exposed by the [`@o3r/eslint-config`](https://www.npmjs.com/package/@o3r/eslint-config) package.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+## Rules changes
+
+> [!NOTE]
+> All the rules set to `warn` has been upgraded to `error`.
+
+Below you can fing, the list of differences by plugin between configurations exposed by `@o3r/eslint-config-otter` and `@o3r/eslint-config`.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:13]:
```suggestion
Below, you can find the list of differences between the configurations exposed by `@o3r/eslint-config-otter` and by `@o3r/eslint-config` for each plugin.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+### Angular ESLint rules
+
+#### Added
+
+We added the [recommended rules](https://github.com/angular-eslint/angular-eslint/blob/main/packages/angular-eslint/src/configs/README.md#angular-eslinttemplate-accessibility) for accessibility in the template.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:19]:
```suggestion
We added the [template accessibility recommended rules](https://github.com/angular-eslint/angular-eslint/blob/main/packages/angular-eslint/src/configs/README.md#angular-eslinttemplate-accessibility).
```

File: migration-guides/ESLINT_FLAT_CONFIG.md (hunk 9 above)
Comment (matthieu-crouzet) [reply migration-guides/ESLINT_FLAT_CONFIG.md:6]: In their doc they use the term `Flat Config` with capital F and C

File: migration-guides/ESLINT_FLAT_CONFIG.md (hunk 11 above)
Comment (cpaulve-1A) [reply migration-guides/ESLINT_FLAT_CONFIG.md:19]: @sdo-1A : What do you think for this one, I am not convince by my suggestion ?

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+- [`semi-spacing`](https://eslint.org/docs/latest/rules/semi-spacing) is deprecated in favor of [`@stylistic/semi-spacing`](https://eslint.style/rules/default/semi-spacing).
+- [`space-in-parens`](https://eslint.org/docs/latest/rules/space-in-parens) is deprecated in favor of [`@stylistic/space-in-parens`](https://eslint.style/rules/default/space-in-parens).
+- [`space-unary-ops`](https://eslint.org/docs/latest/rules/space-unary-ops) is deprecated in favor of [`@stylistic/space-unary-ops`](https://eslint.style/rules/default/space-unary-ops).
+- [`spaced-comment`](https://eslint.org/docs/latest/rules/spaced-comment) is deprecated in favor of [`@stylistic/spaced-comment`](https://eslint.style/rules/default/spaced-comment).
+- [`wrap-iife`](https://eslint.org/docs/latest/rules/wrap-iife) is deprecated in favor of [`@stylistic/wrap-iife`](https://eslint.style/rules/default/wrap-iife).
+
+#### Modified
+
+- [`no-console`](https://eslint.org/docs/latest/rules/no-console) was allowing some `console` methods, now set to `error` without accepting any `console` methods.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:76]:
```suggestion
- [`no-console`](https://eslint.org/docs/latest/rules/no-console) was allowing some `console` methods, is now set to `error` and does not accept any `console` methods.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+### Stylistic ESLint rules
+
+Before the rules from [`@stylistic/eslint-plugin-ts`](https://eslint.style/packages/ts) where used, now we use the ones from [`@stylistic/eslint-plugin`](https://eslint.style/packages/default#stylistic-eslint-plugin) as recommended.
+
+
+### Unicorn ESLint rules
+
+Before, we were using only 3 rules from unicorn. Now, we use their [recommended configuration](https://github.com/sindresorhus/eslint-plugin-unicorn?tab=readme-ov-file#rules) with some customizations.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:85]:
```suggestion
Before, we used three rules from unicorn. We now use their [recommended configuration](https://github.com/sindresorhus/eslint-plugin-unicorn?tab=readme-ov-file#rules) with some customization.
```

File: packages/@o3r/eslint-config/src/rules/utils.cjs [hunk 14]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Convert the rules set as `warn` into `error`
+ * @param {import('@typescript-eslint/utils').TSESLint.FlatConfig.Config | import('@typescript-eslint/utils').TSESLint.FlatConfig.ConfigArray} config config to process
+ * @returns {import('@typescript-eslint/utils').TSESLint.FlatConfig.Rules} only the rules upgraded
+ */
+function upgradeWarningRules(config) {
```
Comment (cpaulve-1A) [inline packages/@o3r/eslint-config/src/rules/utils.cjs:6]: Do we want to update their rules with an ng-update script?

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+#### Removed
+
+- [`@angular-eslint/no-host-metadata-property`](https://github.com/angular-eslint/angular-eslint/blob/main/packages/eslint-plugin/docs/rules/no-host-metadata-property.md) is deprecated and was already set to `off`.
+
+### TypeScript ESLint rules
+
+#### Replaced
+
+- [`@typescript-eslint/no-var-requires`](https://typescript-eslint.io/rules/no-var-requires/) is deprecated in favor of the [`@typescript-eslint/no-require-imports`](https://typescript-eslint.io/rules/no-require-imports/).
+- `no-throw-literal` has been renamed into [`@typescript-eslint/only-throw-error`](https://typescript-eslint.io/rules/only-throw-error/).
```
Comment (sdo-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:30]:
```suggestion
- `no-throw-literal` has been renamed to [`@typescript-eslint/only-throw-error`](https://typescript-eslint.io/rules/only-throw-error/).
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+#### Removed
+
+- [`@typescript-eslint/ban-types`](https://typescript-eslint.io/rules/ban-types/) is deprecated.
+
+#### Added
+
+- [`@typescript-eslint/no-empty-object-type`](https://typescript-eslint.io/rules/no-empty-object-type/) partially replace [`@typescript-eslint/ban-types`](https://typescript-eslint.io/rules/ban-types/), set to `off`.
```
Comment (sdo-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:38]:
```suggestion
- [`@typescript-eslint/no-empty-object-type`](https://typescript-eslint.io/rules/no-empty-object-type/) partially replaces [`@typescript-eslint/ban-types`](https://typescript-eslint.io/rules/ban-types/), set to `off`.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+- [`@typescript-eslint/prefer-promise-reject-errors`](https://typescript-eslint.io/rules/prefer-promise-reject-errors/) set to `error`.
+
+#### Modified
+
+- [`@typescript-eslint/no-unused-vars`](https://typescript-eslint.io/rules/no-unused-vars/) configuration has changed. `caughtErrors` is set to `none` and `ignoreRestSiblings` to `true`.
+
+### Basic ESlint rules
+
+A lot of rules has been deprecated in favor of the [TypeScript](https://typescript-eslint.io/) and [Stylistic](https://eslint.style/) rules.
```
Comment (sdo-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:47]:
```suggestion
A lot of rules have been deprecated in favor of the [TypeScript](https://typescript-eslint.io/) and [Stylistic](https://eslint.style/) rules.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md (hunk 12 above)
Comment (sdo-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md]:
```suggestion
- [`no-console`](https://eslint.org/docs/latest/rules/no-console) previously allowed some `console` methods, it is now set to `error` without accepting any `console` methods.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+- [`spaced-comment`](https://eslint.org/docs/latest/rules/spaced-comment) is deprecated in favor of [`@stylistic/spaced-comment`](https://eslint.style/rules/default/spaced-comment).
+- [`wrap-iife`](https://eslint.org/docs/latest/rules/wrap-iife) is deprecated in favor of [`@stylistic/wrap-iife`](https://eslint.style/rules/default/wrap-iife).
+
+#### Modified
+
+- [`no-console`](https://eslint.org/docs/latest/rules/no-console) was allowing some `console` methods, now set to `error` without accepting any `console` methods.
+
+### Stylistic ESLint rules
+
+Before the rules from [`@stylistic/eslint-plugin-ts`](https://eslint.style/packages/ts) where used, now we use the ones from [`@stylistic/eslint-plugin`](https://eslint.style/packages/default#stylistic-eslint-plugin) as recommended.
```
Comment (sdo-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:80]:
```suggestion
Previously, the rules of [`@stylistic/eslint-plugin-ts`](https://eslint.style/packages/ts) were used, now we use those of [`@stylistic/eslint-plugin`](https://eslint.style/packages/default#stylistic-eslint-plugin) as recommended.
```

File: packages/@o3r/eslint-config/src/rules/utils.cjs [hunk 19]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Convert the rules set as `warn` into `error`
```
Comment (sdo-1A) [inline packages/@o3r/eslint-config/src/rules/utils.cjs:2]:
```suggestion
 * Convert the rules set as `warn` to `error`
```

File: packages/@o3r/eslint-config/src/rules/utils.cjs [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+ * @param {import('@typescript-eslint/utils').TSESLint.FlatConfig.Config | import('@typescript-eslint/utils').TSESLint.FlatConfig.ConfigArray} config config to process
```
Comment (sdo-1A) [inline packages/@o3r/eslint-config/src/rules/utils.cjs:3]: not clear?

File: packages/@o3r/eslint-config/src/rules/utils.cjs [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+ * @returns {import('@typescript-eslint/utils').TSESLint.FlatConfig.Rules} only the rules upgraded
```
Comment (sdo-1A) [inline packages/@o3r/eslint-config/src/rules/utils.cjs:4]:
```suggestion
 * @returns {import('@typescript-eslint/utils').TSESLint.FlatConfig.Rules} only the upgraded rules
```

File: docs/linter/eslint-plugin/rules/no-inner-html.md [hunk 22]
```diff
@@ -7,7 +7,29 @@ For performance reason, it makes sense for certain application to encourage to u
 
 The linter is dedicated to template files and requires the setup of the [@angular-eslint/template-parser](https://www.npmjs.com/package/@angular-eslint/template-parser).
 
-Example of a ESLint configuration:
+Example of a ESLint Flat configuration:
```
Comment (sdo-1A) [inline docs/linter/eslint-plugin/rules/no-inner-html.md:10]:
```suggestion
Example of an ESLint Flat configuration:
```

**Review by kpanot — APPROVED** [review]

File: packages/@o3r/eslint-config/src/rules/utils.cjs (hunk 20 above)
Comment (matthieu-crouzet) [reply packages/@o3r/eslint-config/src/rules/utils.cjs:3]: The 1st one is for the param name 
The 2nd one is the 1st word of the description

Comment (sdo-1A) [reply packages/@o3r/eslint-config/src/rules/utils.cjs:3]: ah yes sorry it wasn't clear, I will resolve this

File: migration-guides/ESLINT_FLAT_CONFIG.md (hunk 11 above)
Comment (sdo-1A) [reply migration-guides/ESLINT_FLAT_CONFIG.md]: I would make a slight change: `recommended rules for template accessibility`
what do you think?

File: docs/linter/eslint-plugin/rules/yarnrc-package-extensions-harmonize.md [hunk 23]
```diff
@@ -6,7 +6,29 @@ This rule checks the `.yarnrc.yml` file to ensure that each version defined in t
...
+      '@o3r': o3rPlugin
+    }
+    rules: {
+      '@o3r/yarnrc-package-extensions-harmonize': 'error'
+    }
+  }
+]
+```
+
+Example of an ESLint Legacy configuration:
```
Comment (cpaulve-1A) [inline docs/linter/eslint-plugin/rules/yarnrc-package-extensions-harmonize.md:31]:
```suggestion
Example of an ESLint legacy configuration:
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 24]
```diff
@@ -0,0 +1,104 @@
+# Migration to ESLint Flat Configuration
+
+[ESLint RC format](https://eslint.org/docs/latest/use/configure/configuration-files-deprecated) is deprecated in favor of [ESLint FlatConfig](https://eslint.org/blog/2022/08/new-config-system-part-2/).
+To migrate, you can follow their [migration guide](https://eslint.org/docs/latest/use/configure/migration-guide).
+
+On the Otter side, an ESLint flat xonfiguration is exposed by the [`@o3r/eslint-config`](https://www.npmjs.com/package/@o3r/eslint-config) package.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:6]:
```suggestion
On the Otter side, an ESLint flat configuration is exposed by the [`@o3r/eslint-config`](https://www.npmjs.com/package/@o3r/eslint-config) package.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+- [`@typescript-eslint/no-var-requires`](https://typescript-eslint.io/rules/no-var-requires/) is deprecated in favor of [`@typescript-eslint/no-require-imports`](https://typescript-eslint.io/rules/no-require-imports/).
+- `no-throw-literal` has been renamed to [`@typescript-eslint/only-throw-error`](https://typescript-eslint.io/rules/only-throw-error/).
+
+#### Removed
+
+- [`@typescript-eslint/ban-types`](https://typescript-eslint.io/rules/ban-types/) is deprecated.
+
+#### Added
+
+- [`@typescript-eslint/no-empty-object-type`](https://typescript-eslint.io/rules/no-empty-object-type/) partially replaces [`@typescript-eslint/ban-types`](https://typescript-eslint.io/rules/ban-types/), set to `off`.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:38]: Is it now set to off or was it its previous value?

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+- [`@typescript-eslint/prefer-promise-reject-errors`](https://typescript-eslint.io/rules/prefer-promise-reject-errors/) set to `error`.
+
+#### Modified
+
+- [`@typescript-eslint/no-unused-vars`](https://typescript-eslint.io/rules/no-unused-vars/) configuration has changed. `caughtErrors` is set to `none` and `ignoreRestSiblings` to `true`.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:43]:
```suggestion
- [`@typescript-eslint/no-unused-vars`](https://typescript-eslint.io/rules/no-unused-vars/) configuration has changed: `caughtErrors` is set to `none` and `ignoreRestSiblings` to `true`.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+- [`spaced-comment`](https://eslint.org/docs/latest/rules/spaced-comment) is deprecated in favor of [`@stylistic/spaced-comment`](https://eslint.style/rules/default/spaced-comment).
+- [`wrap-iife`](https://eslint.org/docs/latest/rules/wrap-iife) is deprecated in favor of [`@stylistic/wrap-iife`](https://eslint.style/rules/default/wrap-iife).
+
+#### Modified
+
+- [`no-console`](https://eslint.org/docs/latest/rules/no-console) previously allowed some `console` methods, it is now set to `error` and does not accept any `console` methods.
+
+### Stylistic ESLint rules
+
+Previously, the rules of [`@stylistic/eslint-plugin-ts`](https://eslint.style/packages/ts) were used, now we use those of [`@stylistic/eslint-plugin`](https://eslint.style/packages/default#stylistic-eslint-plugin) as recommended.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:80]:
```suggestion
Previously, the rules of [`@stylistic/eslint-plugin-ts`](https://eslint.style/packages/ts) were used. We now use those of [`@stylistic/eslint-plugin`](https://eslint.style/packages/default#stylistic-eslint-plugin) as recommended.
```

File: migration-guides/ESLINT_FLAT_CONFIG.md [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+- [`semi-spacing`](https://eslint.org/docs/latest/rules/semi-spacing) is deprecated in favor of [`@stylistic/semi-spacing`](https://eslint.style/rules/default/semi-spacing).
+- [`space-in-parens`](https://eslint.org/docs/latest/rules/space-in-parens) is deprecated in favor of [`@stylistic/space-in-parens`](https://eslint.style/rules/default/space-in-parens).
+- [`space-unary-ops`](https://eslint.org/docs/latest/rules/space-unary-ops) is deprecated in favor of [`@stylistic/space-unary-ops`](https://eslint.style/rules/default/space-unary-ops).
+- [`spaced-comment`](https://eslint.org/docs/latest/rules/spaced-comment) is deprecated in favor of [`@stylistic/spaced-comment`](https://eslint.style/rules/default/spaced-comment).
+- [`wrap-iife`](https://eslint.org/docs/latest/rules/wrap-iife) is deprecated in favor of [`@stylistic/wrap-iife`](https://eslint.style/rules/default/wrap-iife).
+
+#### Modified
+
+- [`no-console`](https://eslint.org/docs/latest/rules/no-console) previously allowed some `console` methods, it is now set to `error` and does not accept any `console` methods.
```
Comment (cpaulve-1A) [inline migration-guides/ESLINT_FLAT_CONFIG.md:76]:
```suggestion
- [`no-console`](https://eslint.org/docs/latest/rules/no-console) previously allowed some `console` methods. It is now set to `error` and does not accept any `console` methods.
```

**Review by sdo-1A — APPROVED** [review]

**Review by cpaulve-1A — APPROVED** [review]

**Pull request merged**